lib/src/models/comment_impl.dart

```diff
@@ -139,25 +158,172 @@ class MoreComments extends RedditBase {
   }
 }
 
-class Comment extends CommentRef {
+/// A fully initialized class which represents a single Reddit comment.
+class Comment extends CommentRef
+    with
+        EditableMixin,
+        GildableMixin,
+        InboxToggleableMixin,
+        InboxableMixin,
+        RedditBaseInitializedMixin,
+        ReplyableMixin,
+        ReportableMixin,
+        SaveableMixin,
+        UserContentMixin,
+        VoteableMixin {
   // CommentModeration get mod; // TODO(bkonyi): implement
 
+  /// Has this [Comment] been approved.
+  bool get approved => data['approved'];
+
+  /// When this [Comment] was approved.
+  ///
+  /// Returns `null` if this [Comment] has not been approved.
+  DateTime get approvedAtUtc => (data['approved_at_utc'] == null)
```

Comment:
I may be sacrificing correctness for code readability, but would be better to have:
```
approved ? new DateTime.fromMillisecondsSinceEpoch(
           data['approved_at_utc'].round() * 1000,
           isUtc: true) : null;
```
Since we aren't checking to see if, data['approved_at_utc'] this might throw, so is it bad practice to rely on external logic (As we are here by relying on approved being true implying that 'approved_at_utc' will be a populated field) to not but the code execution into an internal exception.

Reply:
That does look a bit cleaner, but the best thing to do here is probably just make a helper method (we do this type of operation a lot). Good catch.

Reply:
Added helper.